lms/views/api/canvas/authorize.py

```diff
@@ -12,107 +12,106 @@
 from urllib.parse import urlencode, urlparse, urlunparse
 
 from pyramid.httpexceptions import HTTPFound, HTTPInternalServerError
-from pyramid.view import exception_view_config, view_config, view_defaults
+from pyramid.view import exception_view_config, view_config
 
 from lms.services import CanvasAPIServerError
 from lms.validation.authentication import BearerTokenSchema, CanvasOAuthCallbackSchema
 
-
-@view_defaults(request_method="GET", route_name="canvas_oauth_callback")
-class CanvasAPIAuthorizeViews:
-
-    # The Canvas API scopes that we need for our Canvas Files feature.
-    files_scopes = (
-        "url:GET|/api/v1/courses/:course_id/files",
-        "url:GET|/api/v1/files/:id/public_url",
-    )
-
-    # The Canvas API scopes that we need for our Sections feature.
-    sections_scopes = (
-        "url:GET|/api/v1/courses/:id",
-        "url:GET|/api/v1/courses/:course_id/sections",
-        "url:GET|/api/v1/courses/:course_id/users/:id",
-    )
-
-    def __init__(self, request):
-        self.request = request
-        self.ai_getter = self.request.find_service(name="ai_getter")
-        self.course_service = self.request.find_service(name="course")
-
-    @view_config(permission="canvas_api", route_name="canvas_api.authorize")
-    def authorize(self):
-        authorize_url = urlunparse(
-            (
-                "https",
-                urlparse(self.ai_getter.lms_url()).netloc,
-                "login/oauth2/auth",
-                "",
-                urlencode(
-                    {
-                        "client_id": self.ai_getter.developer_key(),
-                        "response_type": "code",
-                        "redirect_uri": self.request.route_url("canvas_oauth_callback"),
-                        "state": CanvasOAuthCallbackSchema(self.request).state_param(),
-                        "scope": " ".join(self._required_scopes()),
-                    }
-                ),
-                "",
-            )
+#: The Canvas API scopes that we need for our Canvas Files feature.
+FILES_SCOPES = (
+    "url:GET|/api/v1/courses/:course_id/files",
+    "url:GET|/api/v1/files/:id/public_url",
+)
+
+#: The Canvas API scopes that we need for our Sections feature.
+SECTIONS_SCOPES = (
+    "url:GET|/api/v1/courses/:id",
+    "url:GET|/api/v1/courses/:course_id/sections",
+    "url:GET|/api/v1/courses/:course_id/users/:id",
+)
+
+
+@view_config(
+    request_method="GET", route_name="canvas_api.authorize", permission="canvas_api"
```

Comment:
Some of these view config params are duplicated on the different functions now that we can no longer use `@view_defaults`. Things are less confusing as a result, though